spring-boot/legacy-starter/src/test/java/org/togglz/spring/boot/legacy/actuate/BaseTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.togglz.spring.boot.legacy.actuate;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.After;
+import org.springframework.boot.actuate.autoconfigure.ManagementServerPropertiesAutoConfiguration;
+import org.springframework.boot.test.util.EnvironmentTestUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
+import org.togglz.core.Feature;
+import org.togglz.core.activation.Parameter;
+import org.togglz.core.manager.EnumBasedFeatureProvider;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.spi.ActivationStrategy;
+import org.togglz.core.spi.FeatureProvider;
+import org.togglz.core.user.FeatureUser;
+import org.togglz.spring.boot.autoconfigure.TogglzAutoConfiguration;
+import org.togglz.spring.boot.legacy.actuate.autoconfigure.TogglzEndpointAutoConfiguration;
+import org.togglz.spring.boot.legacy.actuate.autoconfigure.TogglzManagementContextConfiguration;
+
+public class BaseTest {
+
```

Comment:
Why don't you make use of `@SpringBootTest`? In general I'm no fan of having a base class for Tests which is inherited by others since you create some kind of dependency to all tests which might have unwanted side effects on change.

Reply:
I understand your point of view, but each test requires a different configuration, and using `@SpringBootTest` would not be practical. In the Spring Boot 2, it's not hard to remove the `BaseTest` class because it doesn't need to maintain any state, but in Spring Boot 1, is not as simple, because it manages the lifecycle of the `ApplicationContext`.

Maybe we can refactor these tests in a future PR?